bowling/Game.cpp

```diff
@@ -77,6 +77,58 @@ Game::Status Game::getGameStatus() const {
     return Game::Status::NO_GAME;
 }
 
+size_t Game::countFramesWithoutStrikeOrSpare(std::vector<Frame>& rolls) {
+    size_t totalPointsWithoutStrikeNorSpare = 0;
+    for (size_t i = 0; i < rolls.size(); i++) {
+        if (!isStrike(rolls[i].getFirstRoll()) && !isSpare(rolls[i].getSecondRoll())) {
+            totalPointsWithoutStrikeNorSpare += (rolls[i].getFirstRoll() + rolls[i].getSecondRoll());
+        } 
+    }
+    return totalPointsWithoutStrikeNorSpare;
+}
+
+size_t Game::countOnlyStrikeFrames(std::vector<Frame>& rolls) {
+    size_t totalOnlyStrikePoints = 0;
+    for (size_t i = 0; i < rolls.size(); i++) {
+        if (isStrike(rolls[i].getFirstRoll())) {
+            totalOnlyStrikePoints += 10;
+            if ((i + 1) != rolls.size()) {
+                if (isStrike(rolls.at(i + 1).getFirstRoll()) || isSpare(rolls.at(i + 1).getSecondRoll())) {
+                    totalOnlyStrikePoints += 10;
+                } else {
+                    totalOnlyStrikePoints += (rolls[i + 1].getFirstRoll() + rolls[i + 1].getSecondRoll());
+                }
+            }
+        }
+    }
+    return totalOnlyStrikePoints;
+}
+
+size_t Game::countOnlySpareFrames(std::vector<Frame>& rolls) {
+    size_t totalOnlySparePoints = 0;
+    for (size_t i = 0; i < rolls.size(); i++) {
+        if (isSpare(rolls[i].getSecondRoll())) {
+            totalOnlySparePoints += 10;
+            if ((i + 1) != rolls.size()) {
+                if (isStrike(rolls.at(i + 1).getFirstRoll()) || isSpare(rolls.at(i + 1).getSecondRoll())) {
```

Comment:
I think there is a mistake here. If we have a spare in current frame, points for this frame are:
10 + points for the next roll
If there is a spare in the next frame, we still only add points scored for the first roll.
I think that the condition should be just:
`isStrike(rolls.at(i + 1).getFirstRoll()) `

Reply:
You are right. I've changed this, thanks.